src/components/DepositWidget/index.tsx

```diff
@@ -352,48 +385,24 @@ const BalancesDisplay: React.FC<BalanceDisplayProps> = ({
 }) => {
   const windowSpecs = useWindowSizes()
 
-  const [search, setSearch] = useState('')
-  const [hideZeroBalances, setHideZeroBalances] = useState(false)
-
-  const handleSearch = (e: React.ChangeEvent<HTMLInputElement>): void => setSearch(e.target.value)
-
-  const handleHideZeroBalances = (e: React.ChangeEvent<HTMLInputElement>): void => setHideZeroBalances(e.target.checked)
-
-  const { value: debouncedSearch, setImmediate: setDebouncedSearch } = useDebounce(search, 500)
-
-  const clearFilters = (): void => {
-    setSearch('')
-    setDebouncedSearch('')
-    setHideZeroBalances(false)
-  }
-
   const [{ localTokens }] = useGlobalState()
 
-  const filteredBalances = useMemo(() => {
-    if ((!debouncedSearch && localTokens.disabled.size === 0) || !balances || balances.length === 0) return balances
-
-    const searchTxt = debouncedSearch.trim().toLowerCase()
-
-    return balances.filter(({ symbol, name, address }) => {
-      if (localTokens.disabled.has(address)) return false
-
-      if (searchTxt === '') return true
-
-      return (
-        symbol?.toLowerCase().includes(searchTxt) ||
-        name?.toLowerCase().includes(searchTxt) ||
-        address.toLowerCase().includes(searchTxt)
-      )
-    })
-  }, [debouncedSearch, balances, localTokens.disabled])
-
-  const displayedBalances = useMemo(() => {
-    if (!hideZeroBalances || !filteredBalances || filteredBalances.length === 0) return filteredBalances
+  const { filteredData: filteredBalances, search, handleSearch } = useDataFilter({
+    data: balances,
+    filterFnFactory: customFilterFnFactory((params: TokenDetails) => localTokens.disabled.has(params.address)),
+    customStopCheck: () => localTokens.disabled.size === 0,
```

Comment:
Can be memoized, so that they don't cause rerenders in [useDataFilter::filteredData](https://github.com/gnosis/dex-react/blob/8ad03b8173febe013d7eda0934fee473c608cd48/src/hooks/useDataFilter.tsx#L53)
```ts
const dataFilterFuncs = useMemo(() => ({
  filterFnFactory: customFilterFnFactory((params: TokenDetails) => localTokens.disabled.has(params.address)),
  customStopCheck: () => localTokens.disabled.size === 0,
}), [localTokens.disabled])
```
or something